.github/workflows/format.yml

```diff
@@ -31,12 +31,18 @@ jobs:
       with:
         # get all history for the diff
         fetch-depth: 0
-    - name: Set up node (20)
-      uses: actions/setup-node@v3
+    - uses: actions/setup-node@v4
       with:
-        node-version: 22.10.0
-    - name: Yarn install
-      run: yarn
+        node-version-file: '.nvmrc'
+        cache: yarn
+        cache-dependency-path: yarn.lock
+    - name: Restore cached node_modules
+      uses: actions/cache@v4
+      id: node_modules
+      with:
+        path: "**/node_modules"
+        key: node_modules-${{ runner.arch }}-${{ runner.os }}-${{ hashFiles('yarn.lock') }}
```

Comment:
Quesiton: I see that we're restoring the cache with this key, but where/when is the cache crated with this key?

Reply:
When there is a cache miss, the contents in the `path` will be added to the cache with the `key` *after* the workflow completes. So in this case, if there's no `node_modules` we can pull from, we'll do a fresh `yarn install --frozen-lockfile`, then after the workflow completes the `node_modules` will be uploaded to the cache.
https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/caching-dependencies-to-speed-up-workflows#cache-hits-and-misses

Reply:
I'll try to explain how this all works in more depth in my PR description